statsd/statsd.go

```diff
@@ -159,9 +160,19 @@ type ClientInterface interface {
 	// Gauge measures the value of a metric at a particular time.
 	Gauge(name string, value float64, tags []string, rate float64) error
 
+	// Gauge measures the value of a metric at a given time.
+	// Even with client side aggregation enabled, there is no aggregation done on a metric
+	// sent using GaugeWithTimestamp: it is written as is in the serialization buffer.
+	GaugeWithTimestamp(name string, value float64, tags []string, rate float64, timestamp time.Time) error
+
```

Comment:
Why only `gauge` and `count` ? `rate`/`set` doesn't make send but we might want to send `timing`, `histogram` and `distribution` with timestamp too, no ?

Reply:
For now, timed metrics ingestion will only be supported for Gauge and Count (DogStatsD server side and backend side).

Reply:
I think we should explain that somewhere, maybe the README or the Go doc string.